### PR TITLE
Print boolean coverage

### PR DESCRIPTION
**Pull request by manycoding**
```
bool_df = pd.DataFrame({"in_stock": np.random.choice(2, 50000), "third_party": np.random.choice(2, 50000, p=[0.9, 0.1]), "promo": np.full(50000, True, dtype=bool)}, dtype=bool)
other_df = pd.DataFrame({"in_stock": np.random.choice(2, 50000), "third_party": np.random.choice(2, 50000, p=[0.1, 0.9]), "promo": np.full(50000, True, dtype=bool)}, dtype=bool)
arche.rules.others.compare_boolean_fields(bool_df, other_df).show()

>>>Boolean Fields:
	third_party_False, third_party_True relative frequencies differ by more than 10%

```
newplot

### CODE REVIEW

File: src/arche/rules/others.py [hunk 1]
```diff
@@ -6,43 +6,44 @@
...
-        result.add_info("No fields to compare")
+        result.add_info("SKIPPED")
         return result
 
-    source_relative_fr = get_bool_relative_frequency(source_bool)
-    target_relative_fr = get_bool_relative_frequency(target_bool)
-    relative_diffs = abs(source_relative_fr - target_relative_fr) * 100
+    lefts = []
+    for c in source_bool:
+        lefts.append(source_bool[c].value_counts(normalize=True).add_prefix(f"{c}_"))
```
Comment (ivankivanov) [inline src/arche/rules/others.py]: In some cases might be useful to report the NaN values for value counts:

`source_bool[c].value_counts(normalize=True, dropna=False)`

Comment (manycoding) [reply src/arche/rules/others.py]: nan is for coverage, in this context nan equals False.

File: src/arche/rules/others.py [hunk 2]
```diff
@@ -6,43 +6,44 @@
...
+    b_ = pd.concat(lefts)
 
-    err_diffs = relative_diffs[(relative_diffs > 10).all(1)]
+    rights = []
+    for c in target_bool:
```
Comment (ejulio) [inline src/arche/rules/others.py]: these 3 lines are repeated, maybe a helper function.

File: src/arche/rules/others.py [hunk 3]
```diff
@@ -6,43 +6,44 @@
...
+        rights.append(target_bool[c].value_counts(normalize=True).add_prefix(f"{c}_"))
+    s_ = pd.concat(rights)
+    bool_covs = pd.DataFrame({"source": b_, "target": s_}).fillna(0) * 100
+    bool_covs.name = "Coverage for boolean fields"
+    result.stats.append(bool_covs)
+
+    difs = (bool_covs["source"] - bool_covs["target"]).abs()
+
+    err_diffs = difs[difs > 10]
```
Comment (ejulio) [inline src/arche/rules/others.py]: This threshold could go in a constant.
Better if some kind of configuration.

File: src/arche/rules/others.py [hunk 4]
```diff
@@ -6,43 +6,44 @@
...
-                f"{err_diffs.index.values} relative frequencies differ "
-                "by more than 10%"
-            ),
-            err_diffs.to_string(),
+            f"{', '.join(err_diffs.index)} relative frequencies differ "
+            "by more than 10%"
         )
 
-    warn_diffs = relative_diffs[((relative_diffs <= 10) & (relative_diffs > 5)).all(1)]
+    warn_diffs = difs[(difs <= 10) & (difs > 5)]
```
Comment (ejulio) [inline src/arche/rules/others.py]: Same here..

**Comment by manycoding** [comment]
@ivankivanov Your version seem better and more compact indeed
Screenshot 2019-04-10 at 14 27 11

File: src/arche/rules/others.py (hunk 3 above)
Comment (manycoding) [reply src/arche/rules/others.py]: Ok, makes sense https://github.com/scrapinghub/arche/issues/56

**Comment by ivankivanov** [comment]
Yes this was how I imagine it.

**Pull request merged**